runner/local/output.go

```diff
@@ -60,6 +65,13 @@ func (o *localOutput) Close() error {
 	return o.f.Close()
 }
 
+func (o *localOutput) Register(namespace, name string) (uri string) {
+	if o.handler != nil {
+		o.uri = o.handler.AddResource(namespace, name, o.AsFile())
```

Comment:
I think it'd be much simpler to say "everything in this tempdir is shared" and that's a good way of specifying intent. If we need more complexity, I'd add it later.